cmd/seed-controller-manager/controllers.go

```diff
@@ -93,24 +93,47 @@ func createSeedConditionUpToDateController(ctrlCtx *controllerContext) error {
 	)
 }
 
-func defaultComponentSettings(ctrlCtx *controllerContext) kubermaticv1.ComponentSettings {
-	return kubermaticv1.ComponentSettings{
-		Apiserver: kubermaticv1.APIServerSettings{
-			DeploymentSettings:          kubermaticv1.DeploymentSettings{Replicas: utilpointer.Int32Ptr(int32(ctrlCtx.runOptions.apiServerDefaultReplicas))},
-			EndpointReconcilingDisabled: utilpointer.BoolPtr(ctrlCtx.runOptions.apiServerEndpointReconcilingDisabled),
-			NodePortRange:               ctrlCtx.runOptions.nodePortRange,
-		},
-		ControllerManager: kubermaticv1.ControllerSettings{
-			DeploymentSettings: kubermaticv1.DeploymentSettings{
-				Replicas: utilpointer.Int32Ptr(int32(ctrlCtx.runOptions.controllerManagerDefaultReplicas)),
-			},
-		},
-		Scheduler: kubermaticv1.ControllerSettings{
-			DeploymentSettings: kubermaticv1.DeploymentSettings{
-				Replicas: utilpointer.Int32Ptr(int32(ctrlCtx.runOptions.schedulerDefaultReplicas)),
-			},
-		},
+func defaultComponentSettings(ctrlCtx *controllerContext, seed *kubermaticv1.Seed) kubermaticv1.ComponentSettings {
+	// Copy default settings.
+	ret := seed.Spec.DefaultComponentSettings
+
+	if replicas := ctrlCtx.runOptions.apiServerDefaultReplicas; replicas != 0 {
+		if ret.Apiserver.Replicas != nil && replicas != int(*ret.Apiserver.Replicas) {
+			ctrlCtx.log.Panicf("cannot preceed due to conflicting settings, cli argument 'api-server-default-replicas'[%d] and 'seedResource.spec.defaultComponentSettings.apiserver.replicas'[%d] do not match", replicas, *ret.Apiserver.Replicas)
```

Comment:
No panic please. Please return an error all the way up the chain.

Also, "proceed" instead of "preceed".

Also we do not use the [] style of putting values in error messages, we use regular parenthesis ("... `foo.bar` (value here) do not match")